lib/evss/disability_compensation_form/data_translation.rb

```diff
@@ -0,0 +1,236 @@
+# frozen_string_literal: true
+
+module EVSS
+  module DisabilityCompensationForm
+    class DataTranslation
+      def initialize(user, form_content)
+        @user = user
+        @form_content = form_content
+      end
+
+      def convert
+        form['claimantCertification'] = true
+        form['applicationExpirationDate'] = application_expiration_date
+        form['directDeposit'] = get_banking_info
+
+        convert_treatments
+        convert_service_info
+        convert_veteran
+
+        @form_content.compact.to_json
+      end
+
+      private
+
+      def form
+        @form_content['form526']
+      end
+
+      def service_info
+        form['serviceInformation']
+      end
+
+      def veteran
+        form['veteran']
+      end
+
+      def convert_service_info
+        convert_service_periods
+        convert_confinements if service_info['confinements'].present?
+        convert_names if service_info['alternateNames'].present?
+        if service_info['reservesNationalGuardService'].present?
+          service_info['reservesNationalGuardService'] = convert_national_guard_service(
+            service_info['reservesNationalGuardService']
+          )
+        end
+      end
+
+      def convert_service_periods
+        service_info['servicePeriods'].map! do |si|
+          {
+            'serviceBranch' => service_branch(si['serviceBranch']),
+            'activeDutyBeginDate' => si['dateRange']['from'],
+            'activeDutyEndDate' => si['dateRange']['to']
+          }
+        end
+      end
+
+      def convert_confinements
+        service_info['confinements'].map! do |ci|
+          {
+            'confinementBeginDate' => ci['confinementDateRange']['from'],
+            'confinementEndDate' => ci['confinementDateRange']['to'],
+            'verifiedIndicator' => ci['verifiedIndicator']
+          }
+        end
+      end
+
+      def convert_names
+        service_info['alternateNames'].map! do |an|
+          {
+            'firstName' => an['first'],
+            'middleName' => an['middle'],
+            'lastName' => an['last']
+          }.compact
+        end
+      end
+
+      def convert_national_guard_service(reserves_service_info)
+        {
+          'title10Activation' => reserves_service_info['title10Activation'],
+          'obligationTermOfServiceFromDate' => reserves_service_info['obligationTermOfServiceDateRange']['from'],
+          'obligationTermOfServiceToDate' => reserves_service_info['obligationTermOfServiceDateRange']['to'],
+          'unitName' => reserves_service_info['unitName'],
+          'unitPhone' => split_phone_number(reserves_service_info['unitPhone']),
+          'inactiveDutyTrainingPay' => reserves_service_info['inactiveDutyTrainingPay']
+        }.compact
+      end
+
+      def convert_veteran
```

Comment:
although it's all veteran data could there be separate methods for phone, address, and homelessness?